File: app/src/main/java/com/jarvis/app/core/MemoryStore.java
package com.jarvis.app.core;

import android.content.Context;
import android.content.SharedPreferences;
import com.jarvis.app.utils.Constants;

public class MemoryStore {

    private final SharedPreferences sharedPreferences;

    public MemoryStore(Context context) {
        this.sharedPreferences = context.getSharedPreferences(Constants.PREFS_NAME, Context.MODE_PRIVATE);
    }

    public void saveData(String key, String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public String getData(String key, String defaultValue) {
        return sharedPreferences.getString(key, defaultValue);
    }
}

File: jarvis_flutter/lib/core/memory_store.dart
import 'package:shared_preferences/shared_preferences.dart';

class MemoryStore {
  late SharedPreferences _prefs;

  Future<void> init() async {
    _prefs = await SharedPreferences.getInstance();
  }

  Future<void> saveData(String key, String value) async {
    await _prefs.setString(key, value);
  }

  String getData(String key, String defaultValue) {
    return _prefs.getString(key) ?? defaultValue;
  }
}
